src/refnux.coffee

```diff
@@ -50,6 +50,15 @@ createStore = (state = {}) ->
         # execute the action
         try
             newval = action state, dispatch
+
+            if newval and typeof(newval.then) is 'function'
+                dispatching = false
+                return newval.then(
+                    (val) -> dispatch(-> val)
+                ,
+                    (err) -> dispatch(-> throw err)
```

Comment:
because dispatch has error logic for functions throwing errors so i thought to re-use it. It's not doing much. What should we do ?